lib/aws/src/autoScalingGroup.ts

```diff
@@ -124,6 +124,12 @@ class ASGRes extends AWSResource<
     );
 
     const scalingacts: AutoScaling.Activities = await this.getScalingActivities();
+
+    if ((scalingacts.length === 0) && (instanceCount === this.expectedInstanceCount)) {
+      log.info("%s setup: nothing to do, no scaling activities seen recently and already at desired capacity", this.rname);
+      return asg;
+    }
+
```

Comment:
There is a comment down below which says

```
// Desired state: ASG status `successful`, and expected instance count.
```

this needs updating, because it's not true anymore. 

Seemingly, the new definition of the desired state is simply "seeing expected instance count".

In theory we don't ever need to look at the last scaling activity.

It's probably nice to leave the code in here, for debugging.

Reply:
> In theory we don't ever need to look at the last scaling activity.

We need it to cover the initial `create` scenario and check if the ASG reached the desired instance count. It's also how Terraform is doing it, see [this](https://github.com/hashicorp/terraform-provider-aws/blob/068ebd8bb5a703e92342ba69406746c993444033/aws/resource_aws_autoscaling_group_waiting.go#L81-L96).

> this needs updating, because it's not true anymore.

I don't think it needs updating since it covers the scenario when the ASG is created.

Reply:
The maybe not so obvious part... it's noteworthy that `scalingacts.length === 0` is hit only when six weeks past since initial create. As documented, scaling activities for the last six weeks are returned -- that is why this new code path can safely be considered to take effect only for upgrades. And that is why you're right, the code path and success criterion for the initial create did _not_ change. Can maybe tune the code comments later to make this more obvious.